src/sentry/static/sentry/app/views/projectFilters.jsx

```diff
@@ -390,11 +396,94 @@ const ProjectFilters = React.createClass({
   },
 
   componentDidUpdate(prevProps) {
-    if (!this.state.loading && !this.state.stats) {
-      this.processStatsData();
+    if (!this.state.loading && !this.state.formattedData) {
+      this.render();
     }
   },
 
+  getStatOpts() {
+    return(
+      this.getProjectFeatures().has('additional-data-filters') ?
+      {
+        'ip-address': 'IP Address',
+        'release-version': 'Release',
+        'error-message': 'Error Message',
+        'browser-extensions': 'Browser Extension',
+        'legacy-browsers': 'Legacy Browser',
+        localhost: 'Localhost',
+        'web-crawlers': 'Web Crawler',
+        'invalid-csp': 'Invalid CSP',
+        cors: 'CORS'
+      } :
+      {'blacklisted':"Filtered Events"}
+    );
+  },
+
+  formatData(rawData) {
+    return Object.keys(this.getStatOpts()).map(stat => {
+      return {
+        data: rawData[stat].map(([x, y]) => {
+          if (y > 0) {
+            this.setState({blankStats: false});
+          }
+          return {x, y};
+        }),
+        label: this.getStatOpts()[stat],
+        statName: stat
+      };
+    });
+  },
+
+  getFilterStats() {
+    let stat_options = Object.keys(this.getStatOpts());
+    let {orgId, projectId} = this.props.params;
+    let statEndpoint = `/projects/${orgId}/${projectId}/stats/`;
+    let query = {
+      since: this.state.querySince,
+      until: this.state.queryUntil,
+      resolution: '1d'
+    };
+    $.when
+      .apply(
+        $,
+        stat_options.map(stat => {
+          let deferred = $.Deferred();
+          this.api.request(statEndpoint, {
+            query: Object.assign({stat: stat}, query),
+            success: deferred.resolve.bind(deferred),
+            error: deferred.reject.bind(deferred)
+          });
+          return deferred;
+        })
+      )
+      .done(
+        function() {
+          let rawStatsData = {};
+          let expected = this.state.expected - 1;
+          // when there is a single request made, this is inexplicably called without being wrapped in an array
+          if(stat_options.length===1){
+            rawStatsData[stat_options[0]] = arguments[0];
+          } else {
+            for (let i = 0; i < stat_options.length; i++) {
+              rawStatsData[stat_options[i]] = arguments[i][0];
+            }
+          }
+
+          this.setState({
+            rawStatsData: rawStatsData,
+            formattedData: this.formatData(rawStatsData),
+            expected: expected,
+            loading: expected > 0
+          });
+        }.bind(this)
```

Comment:
If you use an arrow function here, you don't need to bind `this`.

Reply:
Seems to break things...gonna punt on debugging that for now and keep this as is in the interest of getting this out.

Reply:
wfm